common/buildcraft/builders/snapshot/Blueprint.java

```diff
@@ -97,9 +98,6 @@ public void deserializeNBT(NBTTagCompound nbt) throws InvalidInputDataException
         NBTTagList list = nbt.hasKey("data", Constants.NBT.TAG_LIST) ? nbt.getTagList("data", Constants.NBT.TAG_INT)
             : null;
         int[] serializedData = nbt.hasKey("data", Constants.NBT.TAG_INT_ARRAY) ? nbt.getIntArray("data") : new int[0];
-        if (serializedData == null && list == null) {
-            throw new InvalidInputDataException("Can't read a blueprint with no data!");
-        }
```

Comment:
...Don't remove this.

We catch and handle ````IOException```` later down the line, we don't handle NullPointerExceptions however.